From 388dddb2be28e62d33181b8e610312b047b9f619 Mon Sep 17 00:00:00 2001
From: Lev Kandel <lmakhlis@google.com>
Date: Thu, 7 Nov 2024 15:53:30 +0000
Subject: [PATCH 1/6] Prepare code for breaking change in Protobuf C++ API.

Protobuf 6.30.0 will change the return types of Descriptor::name() and other methods to absl::string_view.
This makes the code work both before and after such a change.
---
 google/cloud/spanner/internal/status_utils.cc | 3 ++-
 4 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/google/cloud/spanner/internal/status_utils.cc b/google/cloud/spanner/internal/status_utils.cc
index 364e2a83c8985..cf01d5da5d108 100644
--- a/google/cloud/spanner/internal/status_utils.cc
+++ b/google/cloud/spanner/internal/status_utils.cc
@@ -14,6 +14,7 @@
 
 #include "google/cloud/spanner/internal/status_utils.h"
 #include "google/cloud/grpc_error_delegate.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
 #include "google/cloud/internal/status_payload_keys.h"
 #include <google/rpc/error_details.pb.h>
 #include <google/rpc/status.pb.h>
@@ -37,7 +38,7 @@
     for (google::protobuf::Any const& any : proto.details()) {
       if (any.UnpackTo(&resource_info)) {
         google::spanner::v1::Session session;
-        auto session_url = "type.googleapis.com/" + session.GetTypeName();
+        auto session_url = absl::StrCat("type.googleapis.com/", session.GetTypeName());
         return resource_info.resource_type() == session_url;
       }
     }
